tools/go-changelog/cmd/changelog-pr-body-check/main.go

```diff
@@ -57,31 +58,23 @@ func main() {
 		Body:  pullRequest.GetBody(),
 	}
 
-	if err := entry.Validate(); err != nil {
-		log.Printf("error parsing changelog entry in %s: %s", entry.Issue, err)
-		switch err.Code {
-		case changelog.EntryErrorNotFound:
-			body := "Oops! It looks like no changelog entry is attached to" +
-				" this PR. Please include a release note block" +
-				" in the PR body, as described in https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/"
-			log.Fatal(body)
-		case changelog.EntryErrorUnknownTypes:
-			unknownTypes := err.Details["unknownTypes"].([]string)
-
-			body := "Oops! It looks like you're using"
-			if len(unknownTypes) == 1 {
-				body += " an"
-			}
-			body += " unknown release-note type"
-			if len(unknownTypes) > 1 {
-				body += "s"
-			}
-			body += " in your changelog entries:"
-			for _, t := range unknownTypes {
-				body += "\n* " + t
+	if errors := entry.Validate(); errors != nil {
+		log.Printf("error parsing changelog entry in %s: %s", entry.Issue, errors)
+		body := "\nOops! Some errors are detected for your changelog entries:"
+		for i, err := range errors {
+			switch err.Code {
+			case changelog.EntryErrorNotFound:
+				body += fmt.Sprintf("\n\n Issue %d: It looks like no changelog entry is attached to this PR. Please include a release note block in the PR body, as described in https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/", i+1)
+			case changelog.EntryErrorUnknownTypes:
+				body += fmt.Sprintf("\n\n Issue %d: unknown changelog types %v \nPlease only use the types listed in https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/.", i+1, err.Details["type"].(string))
```

Comment:
Rather than including each changelog inline here, what would you think about grouping the issues by changelog entry? So something like:

```
Changelog:
\`\`\`release-note:enhancement
compute engine: fixed an issue in google_resource resource 
\`\`\`

Issues:
  - invalid format. Please follow format in https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/#type-specific-guidelines-and-examples.
```

After typing this I realized that we probably currently only support one error per entry... so this would only make sense if we want to support multiple per entry.

This isn't a blocker; let me know what you think!

Reply:
yes, we currently only support one error per entry. Plus, based on the current number of rules, I don't think we're gonna see multiple errors per changelog entry (to clarify, we do support multiple entries per PR though). But I could reform how the errors are reported. I can follow the example you have and make it more clear.